frontend/src/components/SyncEditor/process.ts

```diff
@@ -251,7 +256,7 @@ const process = (
         validate(validators, mappings, resources, validationErrors, syntaxErrors, protectedRanges)
     }
 
-    if (syntaxErrors.length !== 0 && validationErrors.length !== 0 && cachedSecrets && cacheFiltered) {
+    if ((syntaxErrors.length !== 0 || validationErrors.length !== 0) && cachedSecrets && cacheFiltered) {
```

Comment:
@jeswanke This is the crucial part - validation errors are not assessed if there is a syntax error, so this code was not executed in that case.